client/my-sites/preview/main.jsx

```diff
@@ -127,7 +127,8 @@ class PreviewMain extends Component {
 	}
 
 	updateSiteLocation = ( pathname ) => {
-		const externalUrl = this.props.site.URL + ( pathname === '/' ? '' : pathname );
+		const externalUrl =
+			new URL( this.props.site.URL ).origin + ( pathname === '/' ? '' : pathname );
```

Comment:
Even if this looks safe, ideally, `new URL()` should be within a `try/catch` block, because it can throw if it fails to recognize the argument as a proper URL.